mne/beamformer/_lcmv.py

```diff
@@ -39,28 +39,36 @@ def _reg_pinv(x, reg):
     return linalg.pinv(x), d
 
 
-def _setup_picks(picks, info, forward, noise_cov=None):
-    if picks is None:
-        picks = pick_types(info, meg=True, eeg=True, ref_meg=False,
-                           exclude='bads')
-
-    ok_ch_names = set([c['ch_name'] for c in forward['info']['chs']])
+def _setup_picks(info, forward, data_cov=None, noise_cov=None):
+    # get a list of all channel names:
+    fwd_ch_names = set([c['ch_name'] for c in forward['info']['chs']])
+
+    # handle channels from forward model and info:
+    ch_names = [c['ch_name'] for c in info['chs']
+                if c['ch_name'] not in info['bads'] and
+                c['ch_name'] in fwd_ch_names]
+
+    # handle channels from data cov if data cov is not None
+    # Note: data cov is supposed to be None in tf_lcmv
+    if data_cov is not None:
+        ch_names = [c for c in ch_names
+                    if c not in data_cov['bads'] and
```

Comment:
We have a record of which regularization was used, so in principle we could check if it's `empricial` don't worry about it, otherwise do. But I'd say let's be restrictive to start -- ensure they are the same -- and if people hit use cases where they aren't (and their pipelines are not easily adjusted to account for it) then we can relax the restriction.